### PR TITLE
ref(deletions): Refactor error events task

### PR DESCRIPTION
**Pull request by armenzg**
This is in preparation for Issue Platform deletions (#77794).

This is also a follow-up of #78293.

### CODE REVIEW

File: src/sentry/deletions/defaults/group.py [hunk 1]
```diff
@@ -73,25 +78,6 @@ def set_group_and_project_ids(self) -> None:
         self.group_ids = group_ids
         self.project_ids = list(self.project_groups.keys())
 
-    def chunk(self) -> bool:
```
Comment (armenzg) [inline src/sentry/deletions/defaults/group.py:76]: Each class will implement their approached for chunked deletion.

File: src/sentry/deletions/defaults/group.py [hunk 2]
```diff
@@ -146,31 +169,41 @@ def delete_bulk(self, instance_list: Sequence[Group]) -> bool:
...
 
-        group_ids = [group.id for group in instance_list]
+        self.mark_deletion_in_progress(instance_list)
 
+        error_group_ids = [
+            group.id for group in instance_list if group.issue_category == GroupCategory.ERROR
+        ]
         # Tell seer to delete grouping records with these group hashes
-        call_delete_seer_grouping_records_by_hash(group_ids)
+        call_delete_seer_grouping_records_by_hash(error_group_ids)
```
Comment (armenzg) [inline src/sentry/deletions/defaults/group.py:181]: We only contact seer for error groups.

File: tests/sentry/deletions/test_group.py [hunk 3]
```diff
@@ -3,10 +3,10 @@
 from uuid import uuid4
 
 from sentry import nodestore
-from sentry.deletions.defaults.group import EventDataDeletionTask
+from sentry.deletions.defaults.group import ErrorEventsDeletionTask
 from sentry.deletions.tasks.groups import delete_groups
 from sentry.eventstore.models import Event
-from sentry.issues.grouptype import ReplayDeadClickType
+from sentry.issues.grouptype import FeedbackGroup
```
Comment (armenzg) [inline tests/sentry/deletions/test_group.py:9]: The first use case for deletion is going to be User Feedback. Either group type would have worked. This is not even important to change.

File: src/sentry/deletions/defaults/group.py [hunk 4]
```diff
@@ -135,6 +152,12 @@ def delete_dangling_attachments_and_user_reports(self, events: Sequence[Event])
             event_id__in=event_ids, project_id__in=self.project_ids
         ).delete()
 
+    def delete_events_from_snuba(self) -> None:
```
Comment (armenzg) [inline src/sentry/deletions/defaults/group.py:155]: This is the logic from here:
image

**Review by JoshFerge — APPROVED** [review]

**Pull request merged**